pkg/flow/record.go

```diff
@@ -99,9 +137,28 @@ func (b HumanBytes) String() string {
 	return fmt.Sprintf("%.2f MiB", float64(b)/float64(gibi))
 }
 
+func (m *MacAddr) String() string {
+	return fmt.Sprintf("%02x:%02x:%02x:%02x:%02x:%02x", m[0], m[1], m[2], m[3], m[4], m[5])
+}
+
+func (m *MacAddr) MarshalJSON() ([]byte, error) {
+	return []byte("\"" + m.String() + "\""), nil
+}
+
+func (d Direction) MarshalJSON() ([]byte, error) {
+	switch d {
+	case 0:
+		return []byte(`"INGRESS"`), nil
+	case 1:
+		return []byte(`"EGRESS"`), nil
+	default:
+		return []byte(`"UNKNOWN"`), nil
```

Comment:
This should be impossible right ?

In general, I feel it would be better to remove some fields from JSON if input values are invalid than sending a default value that will make the parsing different for every fields.

Reply:
Good catch. I left it there just to compile. Anyway consider that this was just to get a "nicer" demonstration of the flow information in JSON, but this will be removed in the next task, as long as we move to protobuf or any other binary encoding.